### DISCUSSION TITLE
Improvements with PID loop

### DISCUSSION THREAD

**Discussion by abiosbug** [fixerid/sb-projects#4] [Ideas]
Hi,

First of all, thank you for sharing your efforts on automated navigation, your blueprint works amazing!

I did notice however that the rotational corrections are done in fixed increments,  causing either the alignment to overshoot when near (I did adjust the various values as described in your excellent readme) or the alignment to take very long if not yet near the target vectors.

I've been working on a PID controlled approach, found by puzzling pieces together from Reddit. It seems to work pretty well so I was wondering if this could be implemented for the vector alignment. I don't mind having to add 10 more yolo scripts if needed ;-)

Anyway, I'll give it a try and report back here!

Thanks again!

Biosbug

**Comment by fixerid** [comment]
I've tried some analogous alignment techniques similar to PID, the main reason for this behavior is a compromise between the ISAN accuracy limitations and expedient alignment. While in motion ISAN becomes more inaccurate (ie when roating to align), to compensate that I made the alignment work in bursts to minimize the time in motion and only do small increments to minimize the degree of inaccuracy. With every ship having different yaw/pitch profiles those values are my way of allowing users to customize the behavior to their liking. When I tried continuous alignment or even large adjustments, the succeeding readings would progressively get more inaccurate and in my testing lead to more time aligning. 

Please give it a shot I'm sure there's a way to make it work just with my testing it became less fruitful for me to pursue it. Glad you're enjoying the system!
